src/univariate/continuous/semicircle.jl

```diff
@@ -72,4 +72,15 @@ function cdf(d::Semicircle, x::Real)
     end
 end
 
+function rand(rng::AbstractRNG, d::Semicircle)
+    # Idea:
+    # sample polar coodinates r,θ 
+    # of point uniformly distributed on radius d.r half disk
+    # project onto x axis
+    θ = pi*rand(rng)
+    r = d.r*sqrt(rand(rng))
+    x = cos(θ) * r
+    return x
```

Comment:
I think it would be better to use [`cospi`](https://docs.julialang.org/en/v1/base/math/#Base.Math.cospi) here:
```suggestion
    θ = rand(rng) # multiple of π
    r = d.r * sqrt(rand(rng))
    return cospi(θ) * r
```